syntax = "proto3";

package order;

message Order {
    int32 id = 1;
    int32 product_id = 2;
    int32 order_qty = 3;
}

message CreateOrderRequest {
    Order order = 1;
}

message CreateOrderResponse {
    Order order = 1;
}

message ListOrdersRequest {}

message ListOrdersResponse {
    repeated Order orders = 1;
}

service OrderService {
    rpc CreateOrder (CreateOrderRequest) returns (CreateOrderResponse);
    rpc ListOrders (ListOrdersRequest) returns (ListOrdersResponse);
}
